### ISSUE TITLE
[Waiting on BZ checklist] [$500] Search - No username displayed in search for task message

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#33193]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.13-5
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** 
**Slack conversation:** @

## Action Performed:
1. Open the app and login with user A
2. Assign a new task to user B with any title
3. From any other device, login with user B and send message to task created in step 2
4. From user A, assign task created in step 2 to user C
5. From any other device, login with user C and send message to task created in step 2
6. Check in LHN in user A, for message by user C, user C username will be displayed besides the message
7. Open search and check the task report, it will not have username besides the message as it displayed in LHN

## Expected Result:
App should display username besides message for task with multiple members in search as it does in LHN

## Actual Result:
App does not display username besides message for task with multiple members in search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/1304779c-059b-46da-9e08-a4d4ea69efc4

https://github.com/Expensify/App/assets/115492554/9e4d101e-4318-49ab-8b92-4cbfa28b6409

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0101327f233d3a2169</li>
        <li>Upwork Job ID: 1735761982719885312</li>
        <li>Last Price Increase: 2023-12-22</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Contributor | 28070795</li>
<li>paultsimura | Contributor | 28079580</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @shubham1206agra</details>

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?


### What changes do you think we should make in order to solve the problem?

Change to:

```js
result.alternateText = showChatPreviewLine && lastMessageText ? lastMessageText : Localize.translate(preferredLocale, 'report.noActivityYet');
```

https://github.com/Expensify/App/blob/a24f77b27fce2d79302e2076a32461e595684183/src/libs/OptionsListUtils.js#L531

### What alternative solutions did you explore? (Optional)

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
No username displayed in search for task message
### What is the root cause of that problem?

we display last message text for the task report and 
the OptionsListUtils is not consistent with the SidebarUtils options 
https://github.com/Expensify/App/blob/a24f77b27fce2d79302e2076a32461e595684183/src/libs/OptionsListUtils.js#L420-L422

as in the side bar we have the follwoing conditions  https://github.com/Expensify/App/blob/07d64187afe2cd5f26b912ad8e8a84c5d5b2e83c/src/libs/SidebarUtils.ts#L388-L392

### What changes do you think we should make in order to solve the problem?


We need to make both consistent so we need to add this: 


```js
       } else if (lastAction?.actionName !== CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW && lastActorDisplayName && lastMessageTextFromReport) {
            result.alternateText = `${lastActorDisplayName}: ${lastMessageText}`;
        } else {
            result.alternateText = lastMessageTextFromReport.length > 0 ? lastMessageText : Localize.translate(preferredLocale, 'report.noActivityYet');
        }
```

**Comment by NicMendonca** [comment]
@eVoloshchak proposals for you here!

**Comment by NicMendonca** [comment]
bump @eVoloshchak

**Comment by paultsimura** [comment]
@NicMendonca I tried bumping in Slack as well – no response, unfortunately. Same situation on other issues 😢  
https://expensify.slack.com/archives/C01GTK53T8Q/p1703108470058559

**Comment by shubham1206agra** [comment]
@paultsimura's [proposal](https://github.com/Expensify/App/issues/33193#issuecomment-1858476268) looks good to me.

🎀👀🎀 C+ reviewed

**Comment by marcaaron** [comment]
More stuff related to tasks cc @thienlnam 

Proposal looks good. But why do we show the username in the subtitle for tasks? Seems inconsistent to me. So, just double checking that the behavior is correct?

**Comment by paultsimura** [comment]
> why do we show the username in the subtitle for tasks

We already show this username in LHN, so this task is intended to sync task in Search with the task in LHN.

Just to clarify: this is only for tasks that have multiple participants involved (assigned to A, got a message from them, then reassigned to B, and got a message from them).

**Comment by shubham1206agra** [comment]
@thienlnam @marcaaron What is the status here?

**Comment by thienlnam** [comment]
> More stuff related to tasks cc @thienlnam

Proposal looks good. But why do we show the username in the subtitle for tasks? Seems inconsistent to me. So, just double checking that the behavior is correct?

Yeah that behavior looks correct - as mentioned from Paul that subtitle only shows up with multiple participants so it should have the same behavior as a group chat

**Comment by paultsimura** [comment]
Thanks. The PR is ready for review: https://github.com/Expensify/App/pull/33656

**Comment by NicMendonca** [comment]
@shubham1206agra can you accept the offer please? Thanks!

**Comment by NicMendonca** [comment]
@paultsimura you've been paid!

**Comment by shubham1206agra** [comment]
@NicMendonca Accepted

**Comment by NicMendonca** [comment]
@shubham1206agra don't forget the BZ checklist!

**Comment by NicMendonca** [comment]
@shubham1206agra you've been paid 👍

**Comment by NicMendonca** [comment]
@shubham1206agra bump on the BZ checklist!

**Comment by NicMendonca** [comment]
@shubham1206agra bump!

**Comment by shubham1206agra** [comment]
Oops. Sorry for the delay. I will do this right now.

**Comment by shubham1206agra** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@shubham1206agra] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/22734
* [x]  [@shubham1206agra] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/22734/files#r1463404196
* [x]  [@shubham1206agra] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not required
* [x]  [@shubham1206agra] Determine if we should create a regression test for this bug. No
* [x]  [@shubham1206agra] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@NicMendonca] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:
